finetune/siamese_comparison.py

```diff
@@ -0,0 +1,52 @@
+import numpy as np
+
+from finetune.comparison import Comparison
+from finetune.encoding import ArrayEncodedOutput
+from finetune.network_modules import cosine_similarity
+from finetune.base import BaseModel
+import tensorflow as tf
+
+
+class SiameseComparison(Comparison):
+    """
+    Compares two documents to via a siamese network and produces a similarity score (between 0-1).
+
+    :param config: A :py:class:`finetune.config.Settings` object or None (for default config).
+    :param \**kwargs: key-value pairs of config items to override.
+    """
+
+
+    def _text_to_ids(self, pairs, Y=None):
+        """
+        Format comparison examples as a list of IDs
+
+        pairs: Array of text, shape [batch, 2]
+        """
+        arr_0 = BaseModel._text_to_ids(self, [pair[0] for pair in pairs], Y=Y)
```

Comment:
We had to call the grandparent class here, and get the first item of each pair rather than the first item of the batch.